ddtrace/opentelemetry/tracer.go

```diff
@@ -92,6 +95,28 @@ func (t *oteltracer) Start(ctx context.Context, spanName string, opts ...oteltra
 	// we have to record the attributes  locally.
 	// The span operation name will be calculated when it's ended.
 	s := tracer.StartSpan(spanName, ddopts...)
+
+	// Get the current OpenTelemetry baggage.
+	otelBag := otelbaggage.FromContext(ctx)
+	// Get the ddtrace baggage as a map[string]string.
+	ddBag := baggage.All(ctx)
+
+	// Merge the two baggage maps.
+	// If there are conflicts, the OpenTelemetry baggage wins.
+	if len(ddBag) > 0 || otelBag.Len() > 0 {
+		var members []otelbaggage.Member
+		for key, value := range ddBag {
+			member, _ := otelbaggage.NewMember(key, value)
+			members = append(members, member)
+		}
+		members = append(members, otelBag.Members()...)
+		mergedBag, _ := otelbaggage.New(members...)
+		for _, member := range mergedBag.Members() {
+			s.SetBaggageItem(member.Key(), member.Value())
+		}
+		ctx = otelbaggage.ContextWithBaggage(ctx, mergedBag)
+	}
```

Comment:
This is a lot of copying to do on every call to start an otel span. Can we find opportunities to optimize? e.g:
- rather than getting all the dd baggage and all the otel baggage, just check the size of each. If one is empty, just use the other. If neither is empty, then merge the smaller map onto the bigger map. 

Is something like this possible?